i6_models/assemblies/rf/aed_ctc_v1.py

```diff
@@ -0,0 +1,151 @@
+            input_layer=ConformerConvSubsample(
+                in_dim,
+                out_dims=[Dim(32, name="conv1"), Dim(64, name="conv2"), Dim(64, name="conv3")],
+                filter_sizes=[(3, 3), (3, 3), (3, 3)],
+                pool_sizes=[(1, 2)],
+                strides=[(1, 1), (3, 1), (2, 1)],
+            ),
+            encoder_layer_opts=cfg.enc_conformer_layer_opts,
+            num_layers=cfg.num_enc_layers,
+            num_heads=cfg.enc_att_num_heads,
+            dropout=cfg.enc_dropout,
+            att_dropout=cfg.enc_att_dropout,
+        )
+        self.decoder = TransformerDecoder(
+            num_layers=cfg.num_dec_layers,
+            encoder_dim=cfg.enc_model_dim,
+            vocab_dim=target_dim,
+            model_dim=cfg.dec_model_dim,
+        )
+
+        self.target_dim = target_dim
+        self.blank_idx = cfg.blank_idx
+        self.eos_idx = cfg.eos_idx
+        self.bos_idx = cfg.bos_idx  # for non-blank labels; for with-blank labels, we use bos_idx=blank_idx
+
+        if cfg.enc_aux_logits:
+            wb_target_dim = cfg.aux_target_dim
+            if not wb_target_dim:
+                wb_target_dim = target_dim + 1
+            self.wb_target_dim = wb_target_dim
+
+            if target_dim.vocab and not wb_target_dim.vocab:
+                from returnn.datasets.util.vocabulary import Vocabulary
+
+                # Add blank label to existing vocabulary
+                assert cfg.ctc_blank_label
+                assert wb_target_dim.dimension == target_dim.dimension + 1 and cfg.blank_idx == target_dim.dimension
+                vocab_labels = list(target_dim.vocab.labels) + [cfg.ctc_blank_label]
+                wb_target_dim.vocab = Vocabulary.create_vocab_from_labels(
+                    vocab_labels, user_defined_symbols={cfg.ctc_blank_label: cfg.blank_idx}
+                )
+        for i in cfg.enc_aux_logits:
+            setattr(self, f"enc_aux_logits_{i}", rf.Linear(self.encoder.out_dim, wb_target_dim))
+        self.enc_aux_logits = cfg.enc_aux_logits
+
+        self._specaugment_opts = {
+            "steps": config.typed_value("specaugment_steps") or (0, 1000, 2000),
+            "max_consecutive_spatial_dims": config.typed_value("specaugment_max_consecutive_spatial_dims") or 20,
+            "max_consecutive_feature_dims": config.typed_value("specaugment_max_consecutive_feature_dims")
+            or (in_dim.dimension // 5),
+            "num_spatial_mask_factor": config.typed_value("specaugment_num_spatial_mask_factor") or 100,
+        }
+
+    def encode(
+        self, source: Tensor, *, in_spatial_dim: Dim, collected_outputs: Optional[Dict[str, Tensor]] = None
+    ) -> Tuple[rf.State, Dim]:
+        """encode, and extend the encoder output for things we need in the decoder"""
+        source, in_spatial_dim = rf.audio.log_mel_filterbank_from_raw(
+            source, in_spatial_dim=in_spatial_dim, out_dim=self.in_dim, sampling_rate=self.sampling_rate
+        )
```

Comment:
I don't know RF enough yet to know how I would use `i6_models.primitives.feature_extraction.RasrCompatibleLogMelFeatureExtractionV1` here. How would I reconstruct the `in_spatial_dim` that would normally result from the call?